src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -272,10 +271,7 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context, bool l
             PSDriveInfo previousWorkingDrive = CurrentDrive;
 
             // First check to see if the path is a home path
-            if (LocationGlobber.IsHomePath(path))
-            {
-                path = Globber.GetHomeRelativePath(path);
-            }
+            path = Globber.GetHomeRelativePath(path);
```

Comment:
If we're no longer doing a check here, this comment should be updated. 🙂

Reply:
I left this comment as a clarification of intentions.

Reply:
It's somewhat confusing as there is no check performed, the code looks as though we're _assuming_ the path is a home path and proceeding from there.

Am I missing something?